File: 프로그래머스/lv1/천옥수/숫자짝꿍.java
import java.util.*;
class Solution {
    public String solution(String X, String Y) {
        // 공통으로 나타나는 정수 k 를 이용해서 만들 수 있는 가장 큰 정수를 두 수의 짝꿍
        // 존재하지 않으면, 짝꿍 -1

        int[] xList = cntInt(X);
        int[] yList = cntInt(Y);
        System.out.println(Arrays.toString(xList));
        System.out.println(Arrays.toString(yList));
        StringBuilder sb = new StringBuilder();

        for (int i = 0 ; i < 10; i++) {
            if (xList[i] != 0 && yList[i] != 0 ) {
                int iterCnt =Math.min(xList[i], yList[i]);
                for (int j = 0; j < iterCnt ; j++){
                    sb.append(9-i);
                }
            }
        }

        String answer = sb.toString();

        if(answer.equals("")){
            return "-1";
        }
        if(answer.charAt(0) == '0'){
            return "0";
        }
        return answer;
    }

    private int[] cntInt(String x) {
        int[] temp = new int[10];
        for (int i = 0 ; i < x.length() ; i++) {
            temp[9 - (x.charAt(i) - '0')]++;
        }
        return temp;
    }
}

File: 프로그래머스/lv1/숫자짝꿍.cpp
#include <string>
#include <vector>
#include <string>
#include <algorithm>
#include <iostream>
using namespace std;

string solution(string X, string Y) {
    string answer = "";
    vector<int> xCount (10, 0);
    vector<int> yCount (10, 0);
    vector<int> result;
    
    for(int i=0; i<X.length(); i++){
        for(int j=0; j<xCount.size(); j++){
            int temp = X[i] -'0';
            if(temp==j){
                xCount[j]+=1;
            }
        }
    }
    
    for(int i=0; i<Y.length(); i++){
        for(int j=0; j<yCount.size(); j++){
            int temp = Y[i] -'0';
            if(temp==j){
                yCount[j]+=1;
            }
        }
    }
    
    for(int i=9; i>=0; i--){
        int num = min(xCount[i], yCount[i]);
        for(int j=0; j<num; j++)
            answer+=to_string(i);
        }
    if(answer=="") return "-1";
    else if(answer[0]=='0') return "0";
    else return answer;
}
